neo/core/spiketrain.py

```diff
@@ -618,77 +620,118 @@ def time_slice(self, t_start, t_stop):
 
         return new_st
 
-    def merge(self, other):
+    def merge(self, *others):
         '''
-        Merge another :class:`SpikeTrain` into this one.
+        Merge other :class:`SpikeTrain` objects into this one.
 
         The times of the :class:`SpikeTrain` objects combined in one array
         and sorted.
 
-        If the attributes of the two :class:`SpikeTrain` are not
+        If the attributes of the :class:`SpikeTrain` objects are not
         compatible, an Exception is raised.
         '''
-        if self.sampling_rate != other.sampling_rate:
-            raise MergeError("Cannot merge, different sampling rates")
-        if self.t_start != other.t_start:
-            raise MergeError("Cannot merge, different t_start")
-        if self.t_stop != other.t_stop:
-            raise MemoryError("Cannot merge, different t_stop")
-        if self.left_sweep != other.left_sweep:
-            raise MemoryError("Cannot merge, different left_sweep")
-        if self.segment != other.segment:
-            raise MergeError("Cannot merge these two signals as they belong to"
-                             " different segments.")
-        if hasattr(self, "lazy_shape"):
-            if hasattr(other, "lazy_shape"):
-                merged_lazy_shape = (self.lazy_shape[0] + other.lazy_shape[0])
-            else:
-                raise MergeError("Cannot merge a lazy object with a real"
-                                 " object.")
-        if other.units != self.units:
-            other = other.rescale(self.units)
-        wfs = [self.waveforms is not None, other.waveforms is not None]
+        for other in others:
+            if type(other) not in [SpikeTrain, neo.io.proxyobjects.SpikeTrainProxy]:
```

Comment:
It is considered more Pythonic (I'm not sure why), to write:
```
if not isinstance(other, (SpikeTrain, neo.io.proxyobjects.SpikeTrainProxy))
```

(by the way, @samuelgarcia, this code makes me think that `proxyobjects` should be moved to `core`, since code in `neo.core` should not be referring to code in `neo.io`)

Reply:
I agree, at the beginning proxyobject.py and basefromrawio.py was in a separate directory.
I finally put every thing in neo.io.
We can move proxyobject.p > neo.core and keep basefromrawio.py in neo.io.

Reply:
I changed the check to `isinstance`. I agree that `proxyobject.py` should probably not be in `neo.io`, I actually have to do `import neo` in `spiketrain.py` at the moment because importing from `neo.io` causes an import cycle.

Reply:
I will do this in a separate PR.